jerry-core/parser/js/collections/linked-list.cpp

```diff
@@ -286,14 +287,12 @@ linked_list_remove_element (linked_list list, /**< linked list's identifier */
 
   for (size_t i = 0; i < element_num; i++)
   {
+    chunk_prev_to_chunk_with_last_elem_p = list_chunk_iter_p;
     element_iter_p = linked_list_switch_to_next_elem (header_p, &list_chunk_iter_p, element_iter_p);
     JERRY_ASSERT (element_iter_p != NULL);
   }
 
   uint8_t *next_elem_iter_p = linked_list_switch_to_next_elem (header_p, &list_chunk_iter_p, element_iter_p);
-  JERRY_ASSERT (next_elem_iter_p != NULL);
```

Comment:
@wateret, would `(element_num + 1 == list_length && next_elem_iter_p == NULL) || (next_elem_iter_p != NULL)` be correct condition? If so, we could put it to the assert.